### PR TITLE
Consume netguard as a dependency

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203208709178851/f

### Description
Remove netguard submodule and consume it as a library dependency

### Steps to test this PR
- [x] pull this branch and ensure the `vpn-network/vpn-network-impl/src/main/cpp` is deleted
- [x] clean build from this branch
- [x] verify it builds
- [x] install app and smoke test AppTP

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2656** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2656" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2656?utm_source=stack-comment).

File: build.gradle [hunk 1]
```diff
@@ -39,6 +40,10 @@ allprojects {
     repositories {
         google()
         mavenCentral()
+        maven { url 'https://jitpack.io' }
+//        maven {
+//            url = uri("https://maven.pkg.github.com/duckduckgo/netguard")
+//        }
```
Comment (aitorvs) [inline build.gradle:46]: I will delete this in the final version

File: vpn-network/vpn-network-impl/build.gradle [hunk 2]
```diff
@@ -33,6 +33,8 @@ dependencies {
 
     implementation Kotlin.stdlib.jdk7
     implementation "com.squareup.logcat:logcat:_"
+//    implementation "com.duckduckgo.mobile.android:ddg-android-netguard-release:1.0.0"
+    implementation "com.github.duckduckgo:netguard:jitpack-SNAPSHOT"
```
Comment (aitorvs) [inline vpn-network/vpn-network-impl/build.gradle:37]: This is to be changed to version number once https://github.com/duckduckgo/netguard/pull/12 is merged and we create the version tag

File: vpn-network/vpn-network-impl/build.gradle [hunk 3]
```diff
@@ -49,12 +51,6 @@ dependencies {
 }
 
 android {
-    externalNativeBuild {
-
-        cmake {
-            path "CMakeLists.txt"
-        }
-    }
```
Comment (aitorvs) [inline vpn-network/vpn-network-impl/build.gradle:57]: bye bye submodule

**Review by nshuba — APPROVED** [review]
LGTM! Good riddance to the submodule :)

**Pull request merged**